### PR TITLE
Fix no class checks crash

### PR DESCRIPTION
**Pull request by coreyfloyd**
We have another crash in hockey

This is due to making assumptions about the classes we get back from JS

See T139751 and https://rink.hockeyapp.net/manage/apps/152731/app_versions/62/crash_reasons/127780360

### CODE REVIEW

File: Wikipedia/Code/NSDictionary+WMFExtensions.m [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+#import "NSDictionary+WMFExtensions.h"
+
+@implementation NSDictionary (WMFExtensions)
+
+- (BOOL)containsNullObjects {
```
Comment (joewalsh) [inline Wikipedia/Code/NSDictionary+WMFExtensions.m]: should this be recursive? so if a value in a dictionary is also a dictionary, it should return TRUE if that contains NSNull

Comment (coreyfloyd) [inline Wikipedia/Code/NSDictionary+WMFExtensions.m]: @joewalsh I could go either way - you have a preference? Should we have 2 versions - one recursive and one not?

**Comment by joewalsh** [comment]
👍 this is good, maybe with a follow on ticket to recursively remove null

File: Wikipedia/Code/NSDictionary+WMFExtensions.m [hunk 2]
```diff
@@ -42,4 +42,12 @@ - (BOOL)wmf_recursivelyContainsNullObjects{
     return hasNull;
 }
 
+
+- (NSDictionary*)wmf_dictionaryByRemovingNullObjects{
+    return [self bk_reject:^BOOL(id key, id obj) {
+        return [obj isKindOfClass:[NSNull class]];
```
Comment (joewalsh) [inline Wikipedia/Code/NSDictionary+WMFExtensions.m]: a quicker check would be obj == [NSNull null]

File: Wikipedia/Code/WebViewController.m [hunk 3]
```diff
@@ -86,121 +86,154 @@ - (instancetype)initWithSession:(SessionSingleton*)aSession {
...
-            
-            NSString* selectedImageSrcURLString = imageClicked[@"src"];
-            NSCParameterAssert(selectedImageSrcURLString.length);
-            if (!selectedImageSrcURLString.length) {
-                DDLogError(@"Image clicked callback invoked with empty src url: %@", imageClicked);
-                return;
+        }
+    }
+
+    if ([message.body isKindOfClass:[NSString class]]) {
```
Comment (montehurd) [inline Wikipedia/Code/WebViewController.m]: The message "name" checks below should, I think, be part of the message name "if" check above. Formerly the single "if" above was basically functioning like to a "switch" for all the name of messages which could be received. But here we've broken out a second "name if" structure, and inverted the logic, now checking the body before the name.

File: Wikipedia/Code/WMFImageTag.m [hunk 4]
```diff
@@ -84,9 +88,14 @@ - (instancetype)initWithImageTagContents:(NSString*)imageTagContents {
         *stop = width && height && dataFileWidth && dataFileHeight && src && srcAttributeRange.location != NSNotFound;
     }];
 
+    //Don't continue initialization if we have invalid src
+    if([src length] == 0){
```
Comment (montehurd) [inline Wikipedia/Code/WMFImageTag.m]: Should this be doing the same `if([[src stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]] length] == 0){` check?

**Comment by montehurd** [comment]
added one comment. 👍

File: Wikipedia/Code/WMFImageTag.m (hunk 4 above)
Comment (coreyfloyd) [inline Wikipedia/Code/WMFImageTag.m]: We don't have to because its in the default initializer (thats why I added it there so we only need to do it once)

**Pull request merged**